src/main/kotlin/note/application/NoteDeleter.kt

```diff
@@ -15,7 +14,8 @@ class NoteDeleter(
      * @throws NonExistentNoteException if the note does not exist or is not saved on the repository.
      * @see Note
      */
-    fun delete(noteId: Identifier) =
+    fun delete(noteId: String) = delete(Identifier(noteId))
+    internal fun delete(noteId: Identifier) =
```

Comment:
This function is only used on the test suite for the `NoteDeleter` use case:

```kotlin
    @Test
    fun `Invalid identifier throws NonExistentNoteException`() {
        val note = NoteMother.getValidNoteWithDescription()
        val id = NoteMother.getIdentifierFrom(note)
        Mockito.`when`(repository.search(id)).thenReturn(null)

        assertThrows<NonExistentNoteException> {
            noteDeleter.delete(id)
        }

        Mockito.verify(repository, Mockito.times(0)).delete(id)
    }
````

So it should be changed to private and in the test the public function that receives primitives should be tested instead, like this:

```kotlin
    @Test
    fun `Invalid identifier throws NonExistentNoteException`() {
        val note = NoteMother.getValidNoteWithDescription()
        val id = NoteMother.getIdentifierFrom(note)
        Mockito.`when`(repository.search(id)).thenReturn(null)

        assertThrows<NonExistentNoteException> {
            noteDeleter.delete(NoteMother.getIdentifierPrimitiveFrom(note))
        }

        Mockito.verify(repository, Mockito.times(0)).delete(id)
    }
````

Please note that the mother should delegate the primitive retrieval to its corresponding mother, like this:

```kotlin
    fun getIdentifierPrimitiveFrom(note: Note): String = IdentifierMother.getPrimitiveFrom(note.id)

```